src/main/java/duke/UserInterface.java

```diff
@@ -0,0 +1,232 @@
+
+    /**
+     * Executes the command give.
+     * If command given is invalid, user will be notified through standard output.
+     * 
+     * @param userCommand The command given by the user.
+     */
+    public static void executeCommand(Command userCommand) {
+        if (userCommand == Command.EXIT) {
+            showExitMessage();
+            Duke.isRunning = false;
+        } else if (userCommand == Command.LIST) {
+            showList();
+        }
+
+        if (userCommand == Command.ADD_TODO || userCommand == Command.ADD_DEADLINE
+                || userCommand == Command.ADD_EVENT) {
+            try {
+                addTask(userCommand, userInputs[REMAINING_USER_INPUT_INDEX]);
+            } catch (ArrayIndexOutOfBoundsException e) {
+                showWrongFormat();
+            }
+        } else if (userCommand == Command.DONE) {
+            setDone();
+        } else if (userCommand == Command.DELETE) {
+            deleteTask();
+        } else if (userCommand == Command.INVALID) {
+            showInvalidCommand();
+        }
+    }
+
+    private static void deleteTask() {
+        try {
+            int taskIndex = getTaskIndex(userInputs[REMAINING_USER_INPUT_INDEX]);
+            showItemDeleted(TaskManager.delete(taskIndex));
+        } catch (DukeInvalidTaskIndexException | NumberFormatException e) {
+            showInvalidIndex();
+        } catch (ArrayIndexOutOfBoundsException e) {
+            showWrongFormat();
+        }
+    }
+
+    private static void setDone() {
+        try {
+            int taskIndex = getTaskIndex(userInputs[REMAINING_USER_INPUT_INDEX]);
+            TaskManager.setDone(taskIndex);
+            showItemSetDone(taskIndex);
+        } catch (DukeInvalidTaskIndexException | NumberFormatException e) {
+            showInvalidIndex();
+        } catch (ArrayIndexOutOfBoundsException e) {
+            showWrongFormat();
+        }
+    }
+
+    /**
+     * Interpret the command given by the user.
+     * 
+     * @return The command given by the user.
+     */
+    public static Command interpretUserInput() {
+        String userInput = sc.nextLine();
+        if (userInput.replaceAll(" ", "").equals("bye")) {
+            return Command.EXIT;
+        } else if (userInput.replaceAll(" ", "").equals("list")) {
+            return Command.LIST;
+        }
+
+        userInputs = splitCommandAndRemainder(userInput);
+
+        if ("todo".equals(userInputs[USER_COMMAND_INDEX])) {
+            return Command.ADD_TODO;
+        } else if ("deadline".equals(userInputs[USER_COMMAND_INDEX])) {
+            return Command.ADD_DEADLINE;
+        } else if ("event".equals(userInputs[USER_COMMAND_INDEX])) {
+            return Command.ADD_EVENT;
+        } else if ("done".equals(userInputs[USER_COMMAND_INDEX])) {
+            return Command.DONE;
+        } else if ("delete".equals(userInputs[USER_COMMAND_INDEX])) {
+            return Command.DELETE;
+        }
```

Comment:
Consider changing the different types of command e.g. todo, deadline into constants, you may need to use this constants elsewhere as well

Reply:
Thank you. Will change it in future commits.